button-card.js

```diff
@@ -243,33 +304,64 @@
     }
 
     _toggle(state, config) {
-      switch (config.action) {
-        case 'toggle':
-          this.hass.callService('homeassistant', 'toggle', {
-            entity_id: state.entity_id,
-          });
-          break;
-        case 'more_info': {
-          const node = this.shadowRoot;
-          const options = {};
-          const detail = { entityId: state.entity_id };
-          const event = new Event('hass-more-info', {
-            bubbles: options.bubbles === undefined ? true : options.bubbles,
-            cancelable: Boolean(options.cancelable),
-            composed: options.composed === undefined ? true : options.composed,
-          });
-          event.detail = detail;
-          node.dispatchEvent(event);
-          return event;
+      if (!config.tap_action)
+        config.tap_action = 'toggle'
```

Comment:
this is wrong. you want to set the `action` of `tap_action`
you should also not be mutating the passed argument.
instead we should set config to have a default tap_action if one is not defined in `setConfig`.
I'll add a comment there